src/video/kmsdrm/SDL_kmsdrmvideo.c

```diff
@@ -75,37 +72,40 @@ static int
 get_driindex(void)
 {
     int available = -ENOENT;
-    char device[512] = KMSDRM_DRI_PATH;
+    char device[32];
     int drm_fd;
     int i;
     int devindex = -1;
-    DIR *folder = opendir(device);
+    DIR *folder;
+
+    SDL_strlcpy(device, kmsdrm_dri_path, kmsdrm_dri_pathsize + 1);
```

Comment:
The final argument should be the size of the destination buffer, e.g. `sizeof(device)`

Reply:
Or you can make it the maximum size of the source if you know source is never larger than destination, but let's do it your way.